homeassistant/components/mastodon/config_flow.py

```diff
@@ -42,6 +43,11 @@
 )
 
 
+def base_url_from_url(url: str) -> str:
+    """Return the base url from a url."""
+    return urljoin(url, "/").rstrip("/")
```

Comment:
Instead url this, and the stripping, you could also pass it to YARL and get back the origin.

```py
return URL(url).origin()
```

Reply:
Thanks for the suggestion, that's much cleaner